corehq/apps/users/views/mobile/users.py

```diff
@@ -769,8 +770,8 @@ def create_mobile_worker(self, in_data):
             }
 
         self.request.POST = form_data
-        is_valid = lambda: self.new_mobile_worker_form.is_valid() and self.custom_data.is_valid()
-        if not is_valid():
+
+        if not self.new_mobile_worker_form.is_valid() and self.custom_data.is_valid():
```

Comment:
```suggestion
        if not (self.new_mobile_worker_form.is_valid() and self.custom_data.is_valid()):
```
`not A or not B` would also be fine.

Also, there's a potential weirdness in the prior implementation too - if the MW form is not valid, then `custom_data.is_valid` is never called.  I'm pretty sure the implementation of that is something like:
```python
def is_valid(self):
    return not self.errors
```
that means that the reference to `custom_data.errors` should trigger validation anyways, so that probably doesn't matter much

Reply:
Thanks for picking up on this, not sure how I missed that! Fixed here: https://github.com/dimagi/commcare-hq/pull/33184/commits/b1626985200778532e8c63e8d44f5af95e9c5770